### ISSUE TITLE
64-bit Linux: socket() returns a 64-bits integer

### ISSUE THREAD

**Issue by bdemontis** [jwbensley/EtherateMT#49]
In the thd_opt structure, sock is defined as an int32_t. The socket() call returns a 64-bits int using a 64-bits Linux.
In functions rem_int_promisc(), set_int_promisc(), get_if_index_by_name() and get_if_name_by_index()  the sock variable is defined as an int32_t instead of an int.

**Comment by jwbensley** [comment]
Thanks for the bug report @bdemontis  - how have you concluded that Linux is returning a 64-bit int for the `socket()` syscall?

I'm using Linux 5.4 on a 64bit system, the `socket()` syscall is defined here:
https://elixir.bootlin.com/linux/v5.4.30/source/net/socket.c

This in turn calls `__sys_socket()`:
https://elixir.bootlin.com/linux/v5.4.30/source/net/socket.c#L1491

This in turns calls `sock_map_fd()`:
https://elixir.bootlin.com/linux/v5.4.30/source/net/socket.c#L411

This function defines and returns an int (when there is no error):
```
int fd = get_unused_fd_flags(flags);
...

newfile = sock_alloc_file(sock, flags, NULL);
if (!IS_ERR(newfile)) {
     fd_install(fd, newfile);
     return fd;
}
```
This means that a call to `socket()` is returning an int, which on my Ubuntu 18 / 64bit Linux 5.4 system, `sizeof(int)` reports the size as 4 bytes.

Have I made a mistake somewhere?

**Comment by bdemontis** [comment]
Hello James,

Sorry for the late answer: I'm working on the confinement continuity plan
for my employer, and it's taking a lot of my time.

it's my mistake: I'm using an experimental distribution developed
internally that has 64-bits ints, so I have to adapt your code quite a bit.
I guess I hadn't slept for a while when I reported that as a bug. Or maybe
I was drunk.

I am really sorry for this, as apparently I made you lose a lot of time for
absolutely no reason.

Bruno de Montis.

Le dim. 5 avr. 2020 à 21:48, James Bensley <notifications@github.com> a
écrit :
